types/styled-system/index.d.ts

```diff
@@ -13,50 +13,61 @@
 
 import * as CSS from "csstype";
 
-export const defaultBreakpoints: string[];
-
-export function cloneFunc(fn: (...args: any[]) => any): (...args: any[]) => any;
 export function get(obj: any, ...paths: Array<string | number>): any;
-export function themeGet(keys: string, fallback?: string): any;
 
-export function is(n: any): boolean;
-export function isObject(n: any): boolean;
-export function num(n: any): boolean;
-export function px(n: any): string;
+export type Scale = object | Array<string | number>;
 
-export function createMediaQuery(n: number | string): string;
+// Preserved to support v4 shim:
+// https://github.com/styled-system/styled-system/blob/master/packages/styled-system/src/index.js#L108
+export interface LowLevelStyleFunctionArguments<N, S> {
+    prop: string;
+    cssProperty?: string;
+    alias?: string;
+    key?: string;
+    transformValue?: (n: N, scale?: S) => any;
+    scale?: S;
+    // new v5 api
+    properties?: string[];
+}
+
+export function style<N = string | number, S = Scale>(
+    // tslint:disable-next-line no-unnecessary-generics
+    args: LowLevelStyleFunctionArguments<N, S>
+): {
+    [cssProp: string]: string;
+};
 
 export interface styleFn {
     (...args: any[]): any;
-    propTypes?: string[];
 }
 
-export type Scale = object | Array<string | number>;
-
-export interface LowLevelStylefunctionArguments<N, S> {
-  prop: string;
-  cssProperty?: string;
-  alias?: string;
-  key?: string;
-  transformValue?: (n: N, scale?: S) => any;
-  scale?: S;
+export interface CustomStyleDefinition {
+    /** The CSS property to use in the returned style object (overridden by `properties` if present). */
+    property?: string;
+    /** an array of multiple properties (e.g. `['marginLeft', 'marginRight']`) to which this style's value will be
+     * assigned (overrides `property` when present). */
+    properties?: string[];
+    /** A string referencing a key in the `theme` object. */
+    scale?: string;
+    /** A fallback scale object for when there isn't one defined in the `theme` object. */
+    defaultScale?: any[];
+    /** A function to transform the raw value based on the scale. */
+    transform?: (value: any, scale?: Scale) => any;
 }
 
-export function style<N = string | number, S = Scale>(
-  // tslint:disable-next-line no-unnecessary-generics
-  args: LowLevelStylefunctionArguments<N, S>
-): { [cssProp: string]: string };
-
-export function compose(
-    ...funcs: Array<(...args: any[]) => any>
-): (...args: any[]) => any;
+export interface CustomStyleDefinitions {
+    /** Property name exposed for use in components */
+    [customStyleName: string]: CustomStyleDefinition | boolean;
+}
 
-export function mapProps(mapper: (...args: any[]) => any): (func: any) => (...props: any[]) => any;
+export function compose(...parsers: styleFn[]): styleFn;
+export function system(styleDefinitions: CustomStyleDefinitions): styleFn;
 
 export interface VariantArgs {
     key?: string;
```

Comment:
Variant function is missing `scale` property, should be of type string. ref: https://styled-system.com/variants/

Reply:
Done